### PR TITLE
Tasks permission fix

### PR DESCRIPTION
**Pull request by toplenboren**
Allow admins and quest creators to access quest tasks

### CODE REVIEW

File: Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs [hunk 1]
```diff
@@ -37,19 +37,24 @@ public GetQuestTasksQueryHandler(Db context, UserManager<ApplicationUser> userMa
...
             var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId,
                 cancellationToken: cancellationToken);
 
             if (user == null)
                 return BaseResponse.Failure<List<TaskAndHintsDTO>>("Internal: request user does not exist.");
 
+            var userIsAdmin = await _userManager.IsInRoleAsync(user, "Admin");
+            var userIsQuestCreator = request.UserId == quest.AuthorId; 
+            
+            if (!quest.IsReadyToReceiveTaskAttempts())
```
Comment (usernamedt) [inline Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: why nested if

Comment (toplenboren) [reply Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: it's like 55-58, and I don't like long predicates

imo, with nested if its much more readable

Comment (usernamedt) [reply Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: you can add a line break, in 55-58 it is more like a forced decision due to separate function call

Comment (usernamedt) [reply Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: 😎

File: Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs [hunk 2]
```diff
@@ -37,20 +37,23 @@ public GetQuestTasksQueryHandler(Db context, UserManager<ApplicationUser> userMa
...
-            if (quest.IsHidden)
+            var userIsAdmin = await _userManager.IsInRoleAsync(user, "Admin");
+            var userIsQuestCreator = request.UserId == quest.AuthorId; 
+            
+            if (!quest.IsReadyToReceiveTaskAttempts() && !userIsAdmin && !userIsQuestCreator)
+            {
+                return BaseResponse.Failure<List<TaskAndHintsDTO>>("Quest is not in active state yet.");
+            }
+
+            if (quest.IsHidden && !userIsAdmin && !userIsQuestCreator)
```
Comment (usernamedt) [inline Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: 😎

**Review by usernamedt — APPROVED** [review]

File: Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs [hunk 3]
```diff
@@ -37,20 +37,23 @@ public GetQuestTasksQueryHandler(Db context, UserManager<ApplicationUser> userMa
...
-
             var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId,
                 cancellationToken: cancellationToken);
 
             if (user == null)
                 return BaseResponse.Failure<List<TaskAndHintsDTO>>("Internal: request user does not exist.");
 
-            if (quest.IsHidden)
+            var userIsAdmin = await _userManager.IsInRoleAsync(user, "Admin");
+            var userIsQuestCreator = request.UserId == quest.AuthorId;
```
Comment (usernamedt) [inline Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs]: userIsadminOrQuestCreator?

File: Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs [hunk 4]
```diff
@@ -44,14 +44,15 @@ public GetQuestTasksQueryHandler(Db context, UserManager<ApplicationUser> userMa
                 return BaseResponse.Failure<List<TaskAndHintsDTO>>("Internal: request user does not exist.");
 
             var userIsAdmin = await _userManager.IsInRoleAsync(user, "Admin");
-            var userIsQuestCreator = request.UserId == quest.AuthorId; 
+            var userIsQuestCreator = request.UserId == quest.AuthorId;
+            var userIsAdminOrQuestCreator = (userIsAdmin || userIsQuestCreator);
```
Comment (usernamedt) [inline Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs:48]: 😎😎😎

**Review by usernamedt — APPROVED** [review]

**Pull request merged**